syntax = "proto3";

package course;

option java_multiple_files = true;
option java_package = "com.stdiscm.course";

message Course {
  string id = 1;
  string code = 2;
  string title = 3;
  string description = 4;
  int32 capacity = 5;
  string term = 6;
  string academic_year = 7;
  string section = 8;
  string assigned_faculty_id = 9;
}

message ListCoursesRequest {}

message ListCoursesResponse {
  repeated Course courses = 1;
}

message ListFacultyCoursesRequest {
  string faculty_id = 1;
}

message GetCourseRequest {
  oneof lookup {
    string id = 1;
    string code = 2;
  }
}

message GetCourseResponse {
  Course course = 1;
}

service CourseService {
  rpc ListCourses (ListCoursesRequest) returns (ListCoursesResponse);
  rpc ListFacultyCourses (ListFacultyCoursesRequest) returns (ListCoursesResponse);
  rpc GetCourse (GetCourseRequest) returns (GetCourseResponse);
}
